__tests__/ncnewsapp.test.js

```diff
@@ -87,20 +87,117 @@ describe('/api/articles', () => {
+    return request(app)
+      .get('/api/articles/1/comments')
+      .expect(200)
+      .then((response) => {
+        const { comments } = response.body;
+        expect(Array.isArray(comments)).toBe(true);
+        comments.forEach((comment) => {
+          expect(comment).toHaveProperty('comment_id');
+          expect(comment).toHaveProperty('author');
+          expect(comment).toHaveProperty('article_id');
+          expect(comment).toHaveProperty('votes');
+          expect(comment).toHaveProperty('created_at');
+          expect(comment).toHaveProperty('body');
+        });
+      });
+  });
 
+test('200: Comments should be served with the most recent comments first. ', () => {
+  return request(app)
+    .get('/api/articles/3/comments')
+    .expect(200)
+    .then((response) => {
+      expect(response.body.comments).toBeSortedBy("created_at", { descending: true });
+    });
+  });
 
-
+  test('GET:404 sends an appropriate error message when given a valid but non-existent id', () => {
+    return request(app)
+      .get('/api/articles/999/comments')
+      .expect(404)
+      .then((response) => {
+        expect(response.body.message).toBe('article does not exist');
+      });
+  });
+  test('GET:400 sends an appropriate error message when given an invalid id', () => {
+    return request(app)
+      .get('/api/articles/banana/comments')
+      .expect(400)
+      .then((response) => {
+        expect(response.body.message).toBe("Bad request");
+      });
+  });
+});
+  describe('POST /api/articles/:article_id/comments', () => {
+  test('POST:201 inserts a new comment to the db and sends the new comment back to the client', () => {
+    const newComment = {
+      username: 'butter_bridge',
+      body: 'This is amazing'
+    };
+    return request(app)
+      .post('/api/articles/1/comments')
+      .send(newComment)
+      .expect(201)
+      .then((response) => {
+        expect(response.body.comment).toEqual(   {
+          comment_id: 19,
+          body: 'This is amazing',
+          article_id: 1,
+          author: 'butter_bridge',
+          votes: 0,
+          created_at: expect.any(String)
+        } );
+      });
+  });
+  test('POST:400 responds with an appropriate error message when provided with a bad username (no name)', () => {
+    return request(app)
+      .post('/api/articles/1/comments')
+      .send({
+        body: 'This is amazing'
+      })
+      .expect(400)
+      .then((response) => {
+        expect(response.body.message).toBe('Bad request');
+      });
+  });
+  test('POST:404 sends an appropriate error message when given a valid but non-existent id', () => {
+      const newComment = {
+        username: 'butter_bridge',
+        body: 'This is amazing'
+      };
+    return request(app)
+      .post('/api/articles/999/comments')
+      .send(newComment)
+      .expect(404)
+      .then((response) => {
+        expect(response.body.message).toBe('article does not exist');
+      });
+  });
```

Comment:
We should add a test for when the article ID is not a number and a 404 for when the username is not a registered user 🙂